factorial.go

```diff
@@ -0,0 +1,17 @@
+package main
+
+import "fmt"
+
+func factorial(i uint) uint {
+	if i == 0 {
+		return 1
+	}
+	var rez, m uint = 1, 0
```

Comment:
You can add a little improvement there. variable m used only in scope of loop. You can just define it in initialisation  section of the loop.

```go
var res uint = 1
for m := 2; m <= i; m++ {
    rez *= m
}
```